Example/Blockstack/ViewController.swift

```diff
@@ -33,25 +38,57 @@ class ViewController: UIViewController {
     func handleSignInSuccess(userData: UserData) {
         print(userData.profile?.name as Any)
         
-        // Read user profile data
-        let retrievedUserData = Blockstack.shared.loadUserData()
-        print(retrievedUserData?.profile?.name as Any)
+        self.updateUI()
         
-        DispatchQueue.main.async {
-            let name: String? = retrievedUserData?.profile?.name ?? "Nameless Person"
-            self.nameLabel?.text = "Hello, \(name!)"
-            self.signInButton?.isHidden = true
+        // Check if signed in
+        // checkIfSignedIn()
+    }
+    
+    @IBAction func signOut(_ sender: Any) {
+        // Sign user out
+        Blockstack.shared.signOut(redirectURI: "myBlockstackApp") { error in
+            if let error = error {
+                print("sign out failed, error: \(error)")
+            } else {
+                print("sign out success")
+            }
         }
-        
+    }
+    
+    @IBAction func putFileTapped(_ sender: Any) {
+        guard let userData = Blockstack.shared.loadUserData(),
+            let privateKey = userData.privateKey,
+            let publicKey = Keys.getPublicKeyFromPrivate(privateKey) else {
+                return
+        }
+
         // Store data on Gaia
-        let content: Dictionary<String, String> = ["property": "value"]
+        let content: Dictionary<String, String> = ["property1": "value", "property2": "hello"]
+        guard let data = try? JSONSerialization.data(withJSONObject: content, options: []),
+            let jsonString = String(data: data, encoding: .utf8) else {
+                return
+        }
+        
+        // Encrypt content
+        guard let cipherText = Encryption.encryptECIES(content: jsonString, recipientPublicKey: publicKey) else {
```

Comment:
`isString` should be set to `true` otherwise decryption doesn't give you back a string.

Can `Encryption.encryptECIES` detect whether or not the `content` was a string like it does on javascript?

Reply:
Once I set `isString` to true, I was successfully able to decrypt the content in using blockstack.js.

Reply:
Got it, fixed. This will be revisited soon as this function should also support an array of bytes as input (like the JS version).

Reply:
Awesome!